src/features/contactmoment/ContactmomentSwitcher.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <prompt-modal
+    :dialog="dialog"
+    message="Let op, je hebt het contactverzoek niet afgerond. Als je van contactmoment wisselt, wordt het contactverzoek niet verstuurd."
+  />
+  <details ref="detailsEl" v-if="contactmomentStore.contactmomenten.length > 1">
+    <summary class="utrecht-button">Contactmomenten</summary>
+    <menu>
+      <li v-for="(moment, idx) in moments" :key="idx">
+        <button
+          :disabled="moment.isCurrent"
+          @click="moment.enable"
+          :class="{ 'is-current': moment.isCurrent }"
+        >
+          <p v-if="moment.isCurrent" class="current-moment">
+            Huidig contactmoment
+          </p>
+          <p class="name">
+            {{ moment.description?.name || "Onbekende klant" }}
+          </p>
+          <p v-if="moment.description?.contact" class="contact">
+            {{ moment.description.contact }}
+          </p>
+        </button>
+      </li>
+    </menu>
+  </details>
+</template>
+<script lang="ts" setup>
+import PromptModal from "@/components/PromptModal.vue";
+import { useContactmomentStore } from "@/stores/contactmoment";
+import { onClickOutside, useConfirmDialog } from "@vueuse/core";
+import { computed, ref } from "vue";
+import { getKlantInfo } from "./helpers";
+
+const contactmomentStore = useContactmomentStore();
+const moments = computed(() =>
+  contactmomentStore.contactmomenten.map((x) => {
```

Comment:
I'd opt to call this `.map((contactmoment) ...` instead of `x`, for readability.